src/main/java/org/mockbukkit/mockbukkit/block/data/RailDataMock.java

```diff
@@ -0,0 +1,54 @@
+package org.mockbukkit.mockbukkit.block.data;
+
+import org.bukkit.Material;
+import org.bukkit.block.data.Rail;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class RailDataMock extends BlockDataMock implements Rail
+{
+
+	/**
+	 * Constructs a new {@link RailDataMock} for the provided {@link Material}.
+	 *
+	 * @param material The material this data is for.
+	 */
+	protected RailDataMock(@NotNull Material material)
+	{
+		super(material);
+	}
+
+	@Override
+	public @NotNull Shape getShape()
+	{
+		return this.get(BlockDataKey.RAIL_SHAPE);
+	}
+
+	@Override
+	public void setShape(@NotNull Shape shape)
+	{
+		this.set(BlockDataKey.RAIL_SHAPE, shape);
+	}
+
+	@Override
+	public @NotNull Set<Shape> getShapes()
+	{
+        return Arrays.stream(Shape.values()).collect(Collectors.toSet());
```

Comment:
I believe the shapes might be material dependent, for say powered rail does not allow turns.

Reply:
That's correct, but those are considered special types of rails and are covered by `RedstoneRail` (This includes Powered Rails and Detector Rails). If you agree, i'll include those in a separate request.